src/reactors/RelayOrderReactor.sol

```diff
@@ -134,28 +70,39 @@ contract RelayOrderReactor is ReactorEvents, ReactorErrors, ReentrancyGuard, IRe
         });
     }
 
-    function transferInputTokens(ResolvedRelayOrder memory order) internal {
-        permit2.permitWitnessTransferFrom(
-            order.toPermit(),
-            order.transferDetails(),
-            order.info.swapper,
-            order.hash,
-            RelayOrderLib.PERMIT2_ORDER_TYPE,
-            order.sig
-        );
-    }
-
     /// @notice validate the relay order fields
     /// @dev Throws if the order is invalid
-    function _validateOrder(RelayOrder memory order) internal pure {
+    function _validateOrder(RelayOrder memory order) private view {
         if (order.info.deadline < order.decayEndTime) {
             revert DeadlineBeforeEndTime();
         }
 
+        if (block.timestamp > order.info.deadline) {
+            revert DeadlinePassed();
+        }
+
         if (order.decayEndTime < order.decayStartTime) {
             revert OrderEndTimeBeforeStartTime();
         }
 
+        if (address(this) != address(order.info.reactor)) {
+            revert InvalidReactor();
+        }
         // TODO: add additional validations related to relayed actions, if desired
     }
+
+    function _handleResolvedOrders(ResolvedRelayOrder[] memory resolvedOrders) private {
+        unchecked {
+            for (uint256 i = 0; i < resolvedOrders.length; i++) {
+                ResolvedRelayOrder memory order = resolvedOrders[i];
+                order.transferInputTokens(permit2); // meh I don't like that you pass the address :/
```

Comment:
Yea kind of annoying, not sure how to get around besides keeping it in the Reactor contract. I don't mind it too much, think this is cleaner.